File: src/cracking_the_coding_interview/chapter_01/exercise_09/java/Solution.java
package cracking_the_coding_interview.chapter_01.exercise_09.java;

import org.jetbrains.annotations.NotNull;

class Solution {

    boolean isRotation(@NotNull String first, @NotNull String second) {
        return isSubstring(first + first, second);
    }

    private boolean isSubstring(String first, String second) {
        return first.contains(second);
    }
}

File: src/cracking_the_coding_interview/chapter_01/exercise_09/kotlin/Solution.kt
package cracking_the_coding_interview.chapter_01.exercise_09.kotlin

class Solution {

    fun isRotation(first: String, second: String) = second.isSubstring(first + first)

    private fun String.isSubstring(text: String): Boolean {
        return text.contains(this)
    }
}